### PR TITLE
nyxt: 3.11.6 -> 3.11.7

### PR DESCRIPTION
**Pull request by Vonfry**
## Description of changes

Changelog: https://nyxt-browser.com/article/release-3.11.7.org

## Highlight
- ~~This pr may break some packages due to quri upgrade,  a widely used package.~~
- ~~quri cannot be updated in lisp module scope due to widely using. We also cannot update the one in nyxt simply. Because It is not only a dependency of nyxt, but also the ones in other nyxt deps, and all lispLibs are propagatedBuildInptus causing there are multilpe quri in the environment.~~
- To use the correct quri, I make a new scope for nyxt.
- The mesa on master is 24.0.7 ~~which is different from the one on  my machine (24.0.6). So I haven't test the nyxt update.~~

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Vonfry** [comment]
Result of `nixpkgs-review pr 316427` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>nyxt</li>
    <li>sbclPackages.history-tree</li>
    <li>sbclPackages.nclasses</li>
    <li>sbclPackages.nfiles</li>
    <li>sbclPackages.njson</li>
    <li>sbclPackages.nkeymaps</li>
    <li>sbclPackages.nsymbols</li>
    <li>sbclPackages.nyxt</li>
    <li>sbclPackages.prompter</li>
  </ul>
</details>

**Comment by 7c6f434c** [comment]
Wait, did quri upgrade actually break anything initially? (Looking at the upstream which seems to add a default-old-behaviour parameter for a corner case)

**Comment by Uthar** [comment]
Hmm - here's a refactor idea. Maybe we should move nyxt out of packages.nix. It doesn't fit anymore - it was removed from quicklisp and is now being developed at a faster phase using tip of trunk libraries. But there's no `buildLispApplication` right now. `browsers/nyxt/default.nix`. It makes buildScript awkward (also in stumpwm).

File: pkgs/development/lisp-modules/packages.nix [hunk 1]
```diff
@@ -264,57 +264,57 @@ let
...
     pname = "njson";
-    version = "1.2.2";
+    version = "1.2.2-unstable-2024-03-25";
     src = pkgs.fetchFromGitHub {
       owner = "atlas-engineer";
       repo = "njson";
-      rev = version;
-      sha256 = "sha256-kw5DD0GJp/TeCiYATBY8GL8UKqYS6Q4j0a0eQsdcZRc=";
+      rev = "3847bc749a6f6eb1103bc21f8ef3b4f6b301e822";
+      sha256 = "sha256-TgEo2xQCIlm5pHRnh2qAW5NZBfqAcxvvQ1w901pkbck=";
```
Comment (drupol) [inline pkgs/development/lisp-modules/packages.nix:274]:
```suggestion
      hash = "sha256-TgEo2xQCIlm5pHRnh2qAW5NZBfqAcxvvQ1w901pkbck=";
```

File: pkgs/development/lisp-modules/packages.nix [hunk 2]
```diff
@@ -264,57 +264,57 @@ let
...
     pname = "nsymbols";
-    version = "0.3.2";
+    version = "0.3.2-unstable-2024-03-25";
     src = pkgs.fetchFromGitHub {
       owner = "atlas-engineer";
       repo = "nsymbols";
-      rev = version;
-      sha256 = "sha256-psk29WEA7Hxgp29oUniBNvI+lyZfMkdpa5A7okc6kKs=";
+      rev = "9e94870b710a6216468de15b5ca2731e752df31a";
+      sha256 = "sha256-1jyh4craawFCosksD0MNlO4O4IxJKQHoxbDeoYALjW8=";
```
Comment (drupol) [inline pkgs/development/lisp-modules/packages.nix:287]:
```suggestion
      hash = "sha256-1jyh4craawFCosksD0MNlO4O4IxJKQHoxbDeoYALjW8=";
```

File: pkgs/development/lisp-modules/packages.nix [hunk 3]
```diff
@@ -264,57 +264,57 @@ let
...
     pname = "nclasses";
-    version = "0.6.1";
+    version = "0.6.1-unstable-2024-03-25";
     src = pkgs.fetchFromGitHub {
       owner = "atlas-engineer";
       repo = "nclasses";
-      rev = version;
-      sha256 = "sha256-foXmaLxMYMFieB2Yd2iPsU4EX5kLXq7kyElqGZ47OgI=";
+      rev = "ad74d75c4eb0f0a626c618b0547709e6b04827ac";
+      sha256 = "sha256-Vh+Hm0C/YGz3MGvSbKLcOl+KQZ/7Dm/IG7twTNWhAPs=";
```
Comment (drupol) [inline pkgs/development/lisp-modules/packages.nix:301]:
```suggestion
      hash = "sha256-Vh+Hm0C/YGz3MGvSbKLcOl+KQZ/7Dm/IG7twTNWhAPs=";
```

File: pkgs/development/lisp-modules/packages.nix [hunk 4]
```diff
@@ -264,57 +264,57 @@ let
...
     pname = "nfiles";
-    version = "1.1.4";
+    version = "1.1.4-unstable-2024-03-25";
     src = pkgs.fetchFromGitHub {
       owner = "atlas-engineer";
       repo = "nfiles";
-      rev = version;
-      sha256 = "sha256-4rhpBErQgZHcwZRblxgiYaUmKalvllSbJjnRteDVH6k=";
+      rev = "a0c5514963c77e7bdf4b5f7ddda20fb1f158daa0";
+      sha256 = "sha256-hJbpOcEF+HmOwM8Uu8+2cW4Dky1vEiLwk6rON2LEzLw=";
```
Comment (drupol) [inline pkgs/development/lisp-modules/packages.nix:313]:
```suggestion
      hash = "sha256-hJbpOcEF+HmOwM8Uu8+2cW4Dky1vEiLwk6rON2LEzLw=";
```

File: pkgs/development/lisp-modules/packages.nix [hunk 5]
```diff
@@ -368,143 +368,155 @@ let
...
-      (swank.overrideAttrs (final: prev: {
+  nyxt-gtk = let
+    scope = self.overrideScope (self: super: {
+      quri = (super.quri.overrideAttrs (final: prev: {
+          version = "20240328-git";
+          src = pkgs.fetchFromGitHub {
+            owner = "fukamachi";
+            repo = "quri";
+            rev = "03ecaf3771561d713e58a9c5c22b4d95a7592527";
+            sha256 = "sha256-YqZFZvGVD2QRkD7wFyLEBHn+5mFVHfhxSkcv/nY1qbU=";
```
Comment (drupol) [inline pkgs/development/lisp-modules/packages.nix:380]:
```suggestion
            hash = "sha256-YqZFZvGVD2QRkD7wFyLEBHn+5mFVHfhxSkcv/nY1qbU=";
```

File: pkgs/development/lisp-modules/packages.nix [hunk 6]
```diff
@@ -368,143 +368,155 @@ let
...
+          cluffer
+          cl-cffi-gtk
+          sqlite
+          # TODO: Remove these overrides after quicklisp updates past the June 2023 release
+          (trivial-clipboard.overrideAttrs (final: prev: {
+            src = pkgs.fetchFromGitHub {
+              owner = "snmsts";
+              repo = "trivial-clipboard";
+              rev = "50b3d3a25717ac78fb1f0517635c3cb1c31c7667";
+              sha256 = "sha256-QVLSkgVZ4r+XYIhor9UX4hVY8CvBE3M3mQdjwrcl8qk=";
```
Comment (drupol) [inline pkgs/development/lisp-modules/packages.nix:438]:
```suggestion
              hash = "sha256-QVLSkgVZ4r+XYIhor9UX4hVY8CvBE3M3mQdjwrcl8qk=";
```

File: pkgs/development/lisp-modules/packages.nix [hunk 7]
```diff
@@ -368,143 +368,155 @@ let
...
+            };}))
+          (cl-gobject-introspection.overrideAttrs (final: prev: {
+            src = pkgs.fetchFromGitHub {
+              owner = "andy128k";
+              repo = "cl-gobject-introspection";
+              rev = "83beec4492948b52aae4d4152200de5d5c7ac3e9";
+              sha256 = "sha256-g/FwWE+Rzmzm5Y+irvd1AJodbp6kPHJIFOFDPhaRlXc=";
```
Comment (drupol) [inline pkgs/development/lisp-modules/packages.nix:445]:
```suggestion
              hash = "sha256-g/FwWE+Rzmzm5Y+irvd1AJodbp6kPHJIFOFDPhaRlXc=";
```

File: pkgs/development/lisp-modules/packages.nix [hunk 8]
```diff
@@ -368,143 +368,155 @@ let
...
+            };}))
+          (cl-webkit2.overrideAttrs (final: prev: {
+            src = pkgs.fetchFromGitHub {
+              owner = "joachifm";
+              repo = "cl-webkit";
+              rev = "66fd0700111586425c9942da1694b856fb15cf41";
+              sha256 = "sha256-t/B9CvQTekEEsM/ZEp47Mn6NeZaTYFsTdRqclfX9BNg=";
```
Comment (drupol) [inline pkgs/development/lisp-modules/packages.nix:452]:
```suggestion
              hash = "sha256-t/B9CvQTekEEsM/ZEp47Mn6NeZaTYFsTdRqclfX9BNg=";
```

**Review by drupol — CHANGES_REQUESTED** [review]

**Comment by Vonfry** [comment]
> Wait, did quri upgrade actually break anything initially?

Yes. If I override quri on the top scope and run nixpkgs-review, 240 packages need to rebuilding and  many of them are broken.

> Hmm - here's a refactor idea. Maybe we should move nyxt out of packages.nix. It doesn't fit anymore - it was removed from quicklisp and is now being developed at a faster phase using tip of trunk libraries. But there's no buildLispApplication right now. It makes buildScript awkward (also in stumpwm).

I agree on a refactor. In fact most nyxt dependencies are different from the quicklisp ones. It is a good idea to move nyxt out of the lisp modules and package it in a seperate file. 
However, there are some issues should be fixed. At first, I don't know why the building process finds the built files in submodules in the nix store path instead of the tmp build path. Another question on `buildLispApplication`.

**Comment by Uthar** [comment]
> I don't know why the building process finds the built files in submodules in the nix store path instead of the tmp build path

Situation: package P depends on D:

Normally D compiles itself first. Fasl and D.asd are written to store. P finds D.asd during build, loads its Fasl and compiles Itself as well. However if P depends on slashy subsystem D/sub, build of P will attempt to compile sources of D/sub and write into immutable store. This appears as if it's finding them there, but they don't exist yet. It's usually a path with -tmp suffix and the error is "permission denied". The reason is that ASDF looks for defsystem form of uncompiled D/sys in D.asd.

Though it could also happen with normal systems, because we're currently removing undeclared .asds after build instead of before...

It may be your situation. Maybe not - if you post build script I will help.

**Comment by Vonfry** [comment]
> Normally D compiles itself first. Fasl and D.asd are written to store. P finds D.asd during build, loads its Fasl and compiles Itself as well. However if P depends on slashy subsystem D/sub, build of P will attempt to compile sources of D/sub and write into immutable store. This appears as if it's finding them there, but they don't exist yet. It's usually a path with -tmp suffix and the error is "permission denied". The reason is that ASDF looks for defsystem form of uncompiled D/sys in D.asd.

Thanks, let me try to find a way to avoid this.

I will try to build nyxt without dependencies in lisp modules in another pr in the future, so close this pr.

By the way, in the unstable branch, fset is broken which is one of the dependencies of nyxt.

**Comment by jian-lin** [comment]
> here's a refactor idea. Maybe we should move nyxt out of packages.nix.

related: https://github.com/NixOS/nixpkgs/issues/302795 by one nyxt developer

**Pull request closed without merging**